marketplaceItems/soloPoint/appResources/appData/resources/js/soloPointTablet.js

```diff
@@ -0,0 +1,130 @@
+/*
+    Solo Point
+    Created by Milad Nazeri on 2019-01-07
+    Copyright 2016 High Fidelity, Inc.
+
+    Distributed under the Apache License, Version 2.0.
+    See the accompanying file LICENSE or http://www.apache.org/licenses/LICENSE-2.0.html
+
+    Point to solo someone to hear them better in a crowd!
+    TABLET JS
+    
+*/
+
+
+// *************************************
+// START UTILITY FUNCTIONS
+// *************************************
+// #region Utilty
+
+var debug = true;
+var FALSE = "false";
+
+// Easy log function checking for message, an object to stringify, and whether it should be enabled or not
+function log(message, object, enabled){
+    if (!debug || enabled === FALSE) {
+        return;
+    }
+
+    var finalMessage;
+
+    finalMessage = "\n\n\t" + message + ":" + "\n\n";
+
+    if (object) {
+        finalMessage += "\n\t\t" + JSON.stringify(object, null, 4) + "\n";
+    }
+
+    print(finalMessage);
+}
+
+// #endregion
+// *************************************
+// END UTILITY FUNCTIONS
+// *************************************
+
+var UPDATE_SOLO = "UPDATE_SOLO";
+var CLEAR_LIST = "CLEAR_LIST";
+
+var avatarList = document.getElementById("avatar-list");
+var soloNames = document.getElementById("solo-names");
+
+
+// Create the button to reset the sololist
+function createButton(){
+    var button = document.createElement("button");
+    button.innerHTML = "Clear the list";
+    button.setAttribute("id", "soloButton");
+    soloNames.appendChild(button);
+    button.addEventListener("click", function(){
+        EventBridge.emitWebEvent(JSON.stringify({
+            type: CLEAR_LIST
+        }));
+    })
+}
+
+
+// Handle incoming tablet messages
+function onScriptEventReceived(message) {
+    print(message);
+    var data;
+    try {
+        data = JSON.parse(message);
+        switch (data.type) {
+            case UPDATE_SOLO:
+                log("in update solo");
+                var buttonSearch = document.getElementById("soloButton");
+                var finalList = data.value.map(function(avatar){
+                    return `<li>${avatar}</li>`;
+                }).join("");
+                
+                if (finalList.length > 0) {
+                    
+                    if (!buttonSearch) {
+                        createButton();
```

Comment:
instead of creating the button and removing it at line 89, you should use CSS to show and hide the button that's already been defined in the HTML.

Reply:
Sounds good.  Would make it cleaner.